sessionspaces/src/permissionables/ldap.rs

```diff
@@ -0,0 +1,42 @@
+use crate::SessionSpaces;
+use ldap3::{LdapConnAsync, Scope, SearchEntry};
+
+pub async fn update_gid(sessions: &mut SessionSpaces) {
+    let (conn, mut ldap) = LdapConnAsync::new("ldap://ldap.diamond.ac.uk")
+        .await
+        .unwrap();
+    ldap3::drive!(conn);
+    let session_names = sessions.keys().cloned().collect::<Vec<_>>();
+    for session_chunk in session_names.chunks(1000) {
+        let mut filter = String::from("(&(objectclass=posixgroup)(|");
+        for cn in session_chunk {
+            filter.push_str(&format!("(cn={})", cn.replace("-", "_")));
+        }
+        filter.push_str("))");
```

Comment:
Do we need to construct this query or can we just remove the `cn=` and get all the posix groups in `Groups.diamond.ac.uk`?

Reply:
We might query many unwanted namespaces?

Reply:
But yeah, if we want to separate ldap and sessions then fetching the everything from ldap makes sense!